### PR TITLE
[PyROOT] Add tutorial for the DeclareCppCallable feature

### PR DESCRIPTION
**Pull request by stwunsch**

### CODE REVIEW

**Comment by stwunsch** [comment]
@eguiraud Regarding silencing warnings: I don't think this a feature we have to implement actively. You can either write `python -W ignore foo.py` or use the `warnings` package from the std. We are raising a proper Python warning!

**Comment by eguiraud** [comment]
It doesn't seem right to have to turn off _all_ python warnings just to be able to use a feature the way it's intended to be used without having ROOT warn about it.

File: tutorials/pyroot/pyroot004_CppCallable.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+## \file
+## \ingroup tutorial_pyroot
+## \notebook -nodraw
+## This tutorial illustrates how PyROOT supports declaring C++ callables from
+## Python callables making the for example usable with RDataFrame.
```
[the remark this answers is not in the document]
Comment (stwunsch) [reply tutorials/pyroot/pyroot004_CppCallable.py]: Fixed!

File: tutorials/pyroot/pyroot004_CppCallable.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+data = ROOT.RDataFrame(4).Define("x", "(float)rdfentry_")\
+                         .Define("xpow2", "CppCallable::pypow(x, 2)")\
+                         .AsNumpy()
+
+for col in sorted(data):
+    print("{}: {}".format(col, data[col]))
+
+# The performance of using a Python callable from C++ is similar to a plain C++
+# callable if we can use numba in the back to compile it. See the callable below
+# to enforce this behaviour.
```
[the remark this answers is not in the document]
Comment (stwunsch) [reply tutorials/pyroot/pyroot004_CppCallable.py]: Fixed!

**Comment by stwunsch** [comment]
@eguiraud I've fixed the text. Regarding the warnings, you can also suppress warnings in a `with` block:

```python
import warnings
with warnings.catch_warnings():
    @ROOT.DeclareCppCallable(...)
    def f():
        pass
```

**Comment by stwunsch** [comment]
Well damn, we need numba on the build nodes. @oshadura Is this possible?

**Comment by stwunsch** [comment]
@oshadura If we want to have numba in this tutorial, we would have to add a `FindNumba.cmake` thingy, to enable/disable the tutorial during configure time. This is quite some overhead ... What do you recommend?

**Comment by eguiraud** [comment]
(note that pandas would require the same treatment in principle)

**Comment by stwunsch** [comment]
We have `find_python_module`, which can be used to protect the tutorial :)

**Comment by stwunsch** [comment]
I'll put it there!

**Comment by stwunsch** [comment]
@oshadura Just found out we have this in PyMVA already in use

**Comment by stwunsch** [comment]
Still, we need to install `numba` (and also `pandas`) in some places.

@Axel-Naumann What do you think?

**Comment by stwunsch** [comment]
Investigating issues with python 3.8

**Comment by Axel-Naumann** [comment]
> Still, we need to install numba (and also pandas) in some places.

We usually do it like that:
- introduce a `-Dfeature=On` (incl handling of "fail-on-missing")
- enable it on all platforms that have package managers installing it

Are we done with the latter?

**Comment by stwunsch** [comment]
@etejedor this is no good to go, right?

**Comment by eguiraud** [comment]
There is the pandas/numba build system check issue, but also @stwunsch and I were thinking about a possible redesign of the feature, so maybe we should wait a little bit before "publishing" the feature?

**Comment by stwunsch** [comment]
We just discussed with @oshadura and @etejedor that we will fix the build system issue in a dedicated PR.
But I agree that we should wait for Thursday's PPP until we discuss the redesign idea.

**Comment by stwunsch** [comment]
Closed in favor of the PR with the refactored feature #5310

**Pull request closed without merging**